pandas/_libs/tslibs/offsets.pyx

```diff
@@ -406,6 +406,15 @@ class _BaseOffset(object):
         # will raise NotImplementedError.
         return get_day_of_month(other, self._day_opt)
 
+    def _validate_n(self, n):
+        try:
+            nint = int(n)
+        except (ValueError, TypeError):
+            raise TypeError('`n` argument must be an integer')
+        if n != nint:
+            raise ValueError('`n` argument must be an integer')
```

Comment:
show the type of what was passed (for 1st)
make this a more informative message (e.g. showing both values) (for 2nd one)